### PR TITLE
AB2-35 AB2-41 importing history and bookmark from ghostery

### PR DESCRIPTION
**Pull request by ravjit-cliqz**

### CODE REVIEW

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/db/BrowserDatabaseHelper.java [hunk 1]
```diff
@@ -2509,5 +2526,98 @@ public void updateForNewTable(ContentValues bookmark) {
...
+            final String url = rowCursor.getString(rowCursor.getColumnIndex("url"));
+            final String title = rowCursor.getString(rowCursor.getColumnIndex("title"));
+            final int parentId = rowCursor.getInt(rowCursor.getColumnIndex("bookmark_folder_id"));
+            final long timeStamp = rowCursor.getLong(rowCursor.getColumnIndex("date"));
+            final ContentValues contentValues = new ContentValues();
+            contentValues.put(Bookmarks.TITLE, title);
+            contentValues.put(Bookmarks.URL, url);
+            contentValues.put(Bookmarks.TYPE, url == null ? 0 : 1);
+            contentValues.put(Bookmarks.PARENT, parentId == -1 ? 1 : parentId + maxId);
+            contentValues.put(Bookmarks.POSITION, Math.pow(2,63)*-1);
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/db/BrowserDatabaseHelper.java]: Replace `Math.pox(2, 63) * -1` with `Long.MIN_VALUE`

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/db/BrowserDatabaseHelper.java [hunk 2]
```diff
@@ -2509,5 +2526,98 @@ public void updateForNewTable(ContentValues bookmark) {
...
+    private void migrateBookmarks(SQLiteDatabase db) {
+        final Cursor rowCursor = db.rawQuery("SELECT * FROM ghostery.bookmarks ORDER BY _id ASC", null);
+        final Cursor maxIdCurosr = db.rawQuery("SELECT MAX(_id) AS maxId FROM bookmarks", null);
+        int maxId = 0;
+        while(maxIdCurosr.moveToNext()) {
+            maxId = maxIdCurosr.getInt(0);
+        }
+        maxIdCurosr.close();
+        while(rowCursor.moveToNext()) {
+            final String url = rowCursor.getString(rowCursor.getColumnIndex("url"));
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/db/BrowserDatabaseHelper.java]: The following is more efficient.

```java
if (!rowCursor.moveToNext()) {
     rowCursorsor.close();
     return;
}
final int urlColIdx = rowCursor.getColumnIndex("url");
...
do {
    final String url = rowCursor.getString(urlColIdx);
    ...
} while (rowCursor.moveToNext());
...
```

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/db/BrowserDatabaseHelper.java [hunk 3]
```diff
@@ -2509,5 +2526,98 @@ public void updateForNewTable(ContentValues bookmark) {
...
+            db.insert(Bookmarks.TABLE_NAME, null, contentValues);
+        }
+        rowCursor.close();
+    }
+
+    private void migrateHistory(SQLiteDatabase db) {
+        final HashMap<GhosteryHistoryItem, ArrayList<Long> > historyItems = new HashMap<>();
+        final Cursor c = db.rawQuery("SELECT * FROM ghostery.history", null);
+        while(c.moveToNext()) {
+            final String url = c.getString(c.getColumnIndex("url"));
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/db/BrowserDatabaseHelper.java]: Same here: please do not ask for the column index in each iteration.

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/db/BrowserDatabaseHelper.java [hunk 4]
```diff
@@ -2509,5 +2526,98 @@ public void updateForNewTable(ContentValues bookmark) {
...
+            historyValues.put(History.DATE_CREATED, creationDate);
+            historyValues.put(History.DATE_MODIFIED, modifiedDate);
+            historyValues.put(History.GUID, guid);
+            historyValues.put(History.IS_DELETED, 0);
+            db.insert(History.TABLE_NAME, null, historyValues);
+            for(long timestamp : timeStamps) {
+                final ContentValues visitsValues = new ContentValues();
+                visitsValues.put(Visits.HISTORY_GUID, guid);
+                visitsValues.put(Visits.VISIT_TYPE, 1);
+                visitsValues.put(Visits.DATE_VISITED, timestamp*1000);
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/db/BrowserDatabaseHelper.java]: I suppose this value is a `long`, replace with `timestamp*1000L`

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/db/GhosteryHistoryItem.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package org.mozilla.gecko.db;
+
+/**
+ * @author Ravjit Uppal
+ */
```
Comment (spacifici) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/db/GhosteryHistoryItem.java]: We do not use `@author` in mozilla, when we create new files we just add the following copyright notice:

```java
/*
 * Copyright © Cliqz 2018
 */
```

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/db/BrowserDatabaseHelper.java (hunk 4 above)
Comment (ravjit-cliqz) [reply mozilla-release/mobile/android/base/java/org/mozilla/gecko/db/BrowserDatabaseHelper.java]: long multiplied by int will result in a long :)

**Pull request merged**